src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -70,9 +70,16 @@ internal GeneratorExecutionContext(Compilation compilation, ParseOptions parseOp
         /// Adds source code in the form of a <see cref="string"/> to the compilation.
         /// </summary>
         /// <param name="hintName">An identifier that can be used to reference this source text, must be unique within this generator</param>
-        /// <param name="source">The source code to be add to the compilation</param>
+        /// <param name="source">The source code to add to the compilation</param>
         public void AddSource(string hintName, string source) => AddSource(hintName, SourceText.From(source, Encoding.UTF8));
 
+        /// <summary>
+        /// Adds source code in the form of a <see cref="StringBuilder"/> to the compilation.
+        /// </summary>
+        /// <param name="hintName">An identifier that can be used to reference this source text, must be unique within this generator</param>
+        /// <param name="source">The source code to add to the compilation. This must not be modified from now on.</param>
```

Comment:
This is where I _really_ want move semantics... I think a stronger assertion is needed, perhaps "This builder must not be further modified: undefined behavior will result."?

Reply:
I added stronger language and put it in summary as well.